Post/paalgo

peakBreaker opened the pull request:
Added blogentry on golang with @palsivertsen

peakBreaker committed:
feat: added cover picture for paal go entry

peakBreaker committed:
enh: added go notes to golang post, still wip

peakBreaker committed:
blog: added text on how I know paal + pics

peakBreaker committed:
added some code snippets

peakBreaker committed:
updated gemfile.lock, vuln dep alert by github

peakBreaker commented:
The blogentry can be read in [evernote](https://www.evernote.com/shard/s683/sh/d7329a72-9c01-41da-a8d2-789b4cae88df/bfaf7336cbc03c5cad09f923663de38c)

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+
+*From alg0001 on [reddit](https://www.reddit.com/r/virginvschad/comments/7rkm5n/the_virgin_windows_peasant_vs_the_chad_gnulinux/)*
+
+Pål joined the Sensario company late 2017 with the goal of building the backend of
```
Mid 2017*

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+- Strings, chars, pointers, mutant for loops
+- Implicit interfaces, structs, maps, slices and arrays
+- Made with concurrency in mind
+- Nice utilities: Defer, multiple returns
```
channels?

peakBreaker replied:
from 
`- Made with concurrency in mind`
to 
`- Made with concurrency in mind; Goroutines and channels`

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+- Made with concurrency in mind
+- Nice utilities: Defer, multiple returns
+
+## Whats in $GOPATH?
```
Go 1.11 introduced modules which allows projects to be located outside GOPATH. I haven't looked at it too much yet. See https://github.com/golang/go/wiki/Modules

peakBreaker replied:
Added link to wiki entry.

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+
+- go fmt is the standard for golang formatting/convention: `$ go fmt myfile.go`
+- Golang scope rules are very simple, everything between {} is a scope
+- The difference between pointers in C and Go is that go pointers (...)
```
TODO

peakBreaker replied:
Removed.

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+
+A great tut on this in [A Tour of Go](https://tour.golang.org/basics/3)
+
+## Golang datatypes
```
Formatting in this chapter is weird when rendered

peakBreaker replied:
changed formatting in chapter to be more in line with rest of entry.

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+  3. Slice utils:
+    -> len(slice)  // length of slice
+    -> cap(slice) // capacity of slice
+    -> append(<slice>, <value>[, <more values>]) // This returns a new instance
```
Careful! This might still point to the same underlying array.
I've created an example to illustrate:
https://play.golang.org/p/4W_lzOf0mui

peakBreaker replied:
Added example to blog entry.

peakBreaker committed:
fix: changes based on pr !1 and fix clipping

On phone devices the page was clipping alot due to long lines.  Have
improved the syntax to fall in line with proper style

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+  * The interface type is a set of method signatures
+  * Interfaces are implemented implicitly, if a datatype has the methods to satisfy the interface, it automatically implements the interface
+  * Syntax for declaring interface: type <MyInterface> interface {/*methods*/}
+  * Describing interfaces:  fmt.Printf("(%v, %T)\n", i, i) // i is an interface
```
`%#v` is also worth mentioning

peakBreaker replied:
This describing interfaces text was just to show that interfaces are just and underlying datastructure with data on value and datatype of the *whatever* implementing the interface.  Omitting a long text about fmt here, but maybe a new chapter?

peakBreaker replied:
Changed to

```
fmt.Printf("(%v, %T)\n", i, i) // i is an interface
```

This will show us that the interface is just a tuple of a value and datatype.
This is well illustrated in the [Tour of Go](https://tour.golang.org/methods/14) example.

```
Some interfaces:
(<nil>, <nil>)
(42, int)
(hello, string)
```

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+building stuff concurrently, namely Goroutines and channels
+
+- Goroutines
+    * Basically a concurrent function
```
No. `go` is a statement for starting a new goroutine.
See https://golang.org/ref/spec#Go_statements

peakBreaker replied:
Yep, changed to:

- Goroutines
  * A syntax for starting [goroutines](https://golang.org/ref/spec#Go_statements)
  * Syntax: go <expression>
  * Goroutines must not return anything, but can interact with channels (see
    below)

palsivertsen commented on _posts/2018-09-07-Golang-with-The-Paal-Gang.markdown:
```diff
@@ -0,0 +1,248 @@
+- Goroutines
+    * Basically a concurrent function
+    * Syntax: go <funcion>()
+    * Goroutines must be void
```
`void` is not a Go concept. I think you mean "no return args".
Goroutines can be started both args and return values:
https://play.golang.org/p/BoagKuOa-w8

peakBreaker replied:
nice, changed to 

  * Goroutines cant catch/handle returns, but can interact with channels (see
    below)

peakBreaker committed:
fixes based on review in !1, update dependencies
